### ISSUE TITLE
WordPress 5.6 compatibility – editor width doesn't change based on full-width/default template

### ISSUE THREAD

**Issue by Aljullu** [woocommerce/storefront#1512]
## To Reproduce
1. Create a new page.
2. In the sidebar, open the _Page Attributes_ panel and switch between the _Default template_ and the _Full Width_.
3. Notice the editor width doesn't change as it used to in WordPress 5.5.

## Screenshots
_WordPress 5.5:_
![Peek 2020-10-21 17-22](https://user-images.githubusercontent.com/3616980/96744012-def44400-13c4-11eb-82a0-cadd03657ffa.gif)

_WordPress 5.6:_
![Peek 2020-10-21 17-27](https://user-images.githubusercontent.com/3616980/96744020-e1569e00-13c4-11eb-899a-4b689020bfc9.gif)

## Expected behavior
The width in the editor should be based on the selected template.

**Comment by Aljullu** [comment]
The issue happens because when `editor.js` is loaded, the settings from `wp.data.select( 'core/editor' ).getEditorSettings()` have `alignWide: false`, even though it should be true ([relevant part of the code](https://github.com/woocommerce/storefront/blob/trunk/assets/js/src/editor.js#L43)).

Seems to be related to: https://github.com/WordPress/gutenberg/issues/21267.

**Comment by nerrad** [comment]
Albert, looking at the code you linked to in Storefront, it looks like Storefront only checks the editor settings when the dom is ready so my guess is that the problem here is Storefront's reliance on `wp.domReady` vs subscribing to the selector change?

I think there were some changes to the rendering process in Gutenberg that is likely affecting the utility of the `wp.domReady` event for this use-case.

**Comment by nerrad** [comment]
It looks like there's more involved here than just the value returned from `wp.data.select( 'core/editor' ).getEditorSettings()`. When I use the following code:

```js
        let pageTemplate = getCurrentPageTemplate();
	// Don't do anything if the theme doesn't declare support for `align-wide`. 
        // todo: this is removed because we can't rely on this initial check in WP 5.6+, looks like we'll need to pass through the supports via some other mechanism.
	// if (!themeSupportsWide()) {
	//	return;
	//}
	wp.data.subscribe(() => {

		if (getCurrentPageTemplate() !== pageTemplate) {
			pageTemplate = getCurrentPageTemplate();
			maybeUpdateEditor();
		}
	});
```

...the dom updates for the blocks as expected but it doesn't look like the styles set by the built `gutenberg-editor.css` will apply as expected for WordPress 5.6 because the layout html has changed with Gutenberg in the editor (some blocks support a "lightBlockWrapper" which removes the `.wp-block` class. Also there are new styles for `.wp-block` hierarchy.  So it looks like we may have to build support for pre WP 5.6 and post WP 5.6 editor.

One thing we can also do is maybe do some [detection of whether a block supports `lightBlockWrapper` or not](https://github.com/WordPress/gutenberg/blob/1f0d0f5d945e93232a20741221b125432a5993ac/packages/block-editor/src/components/block-list/block.js#L137-L139) and adjust the dom manipulation accordingly.

However, I think one of the reasons for this change by GB was to make it "easier" for themes to style in the editor. So it'd probably be worthwhile reading through any documentation on theme changes between WP versions (maybe some dev notes?) and see if that can be used to adjust what Storefront does post WP 5.6.

**Comment by Aljullu** [comment]
Thanks for investigating this @nerrad, I created a work-around so it works with WP 5.6 in #1520, but I'm not really happy with the solution since it seems quite 'fragile' and it might break in future WP updates. I explained my worries with more detail in the PR.

> lightBlockWrapper

I did a quick search and I think only FSE blocks support this feature, so my understanding is that this is not something we need to account yet for WP 5.6 compatibility?

**Comment by nerrad** [comment]
> I did a quick search and I think only FSE blocks support this feature, so my understanding is that this is not something we need to account yet for WP 5.6 compatibility?

Hmm right. I think I jumped to an incorrect conclusion for what I was observing with the Quote block html structure pre WP 5.6 and post WP 5.6.

**Comment by Aljullu** [comment]
As discussed in #1520, instead of trying to fix this feature for WP 5.6, we will completely remove it. See #1525 for a removal PR.
